src/bandb/byte.rs

```diff
@@ -1,33 +1,68 @@
+use std::{num::{ParseIntError, TryFromIntError}, str::FromStr};
+use thiserror::Error;
+use super::bits::Bits;
+
 pub struct Byte {
-    pub value: u8,
+    value: u8,
+}
+
+#[derive(Debug, Error)]
+pub enum ByteConversionError {
+    #[error("Cannot convert byte to digit")]
+    Conversion,
+
+    #[error("Cannot downcast value to u8")]
+    Cast(#[from] TryFromIntError),
+
+    #[error("{0:#}")]
+    Creation(#[from] ByteCreationError)
 }
 
-use crate::bits::Bits;
+#[derive(Debug, Error)]
+pub enum ByteCreationError {
+    #[error("Couldn't parse byte value")]
+    ParseError(#[from] ParseIntError)
+}
 
 impl Byte {
-    pub fn from_bits(bits: &Bits) -> Self {
-        let byte = convert_vec_to_u8(&bits.value);
-        Byte { value: byte }
+    pub fn from_bits(bits: &Bits) -> Result<Self, ByteCreationError> {
+        Self::from_iter(bits.value())
+    }
+
+    pub fn from_iter(iter: &[u8]) -> Result<Self, ByteCreationError> {
+        Ok(Self {
+            value: u8::from_str_radix(
+                iter
+                    .iter()
+                    .map(|&byte| byte.to_string())
+                    .collect::<String>()
+                    .as_str(),
+                2
+            )?
+        })
```

Comment:
I created my own `TryFromIterator` trait and implemented it to the `Byte` struct instead of using `FromIterator` as I expect a `Result<Self, ByteCreationError>` to be returned instead of a raw `Self`, I didn't find a `TryFromIterator` method anywhere in the standard library so I created it myself.